src/commands/sourcemaps/upload.ts

```diff
@@ -113,21 +116,46 @@ export class UploadCommand extends Command {
     return apiConstructor(getBaseIntakeUrl(), this.config.apiKey!)
   }
 
-  private getMatchingSourcemapFiles(): Payload[] {
+  private getMatchingSourcemapFiles = async (): Promise<Payload[]> => {
+    const useGit = this.disableGit === undefined || !this.disableGit
     const sourcemapFiles = glob.sync(buildPath(this.basePath!, '**/*.js.map'))
+    let repositoryData: RepositoryData | undefined
+    if (useGit) {
+      const simpleGit = newSimpleGit()
+      repositoryData = await gitInfos(simpleGit, this.context.stdout, this.repositoryURL)
+    }
 
-    return sourcemapFiles.map((sourcemapPath) => {
-      const minifiedFilePath = getMinifiedFilePath(sourcemapPath)
+    return Promise.all(
+      sourcemapFiles.map(async (sourcemapPath) => {
+        const minifiedFilePath = getMinifiedFilePath(sourcemapPath)
+        let infos: string | undefined
```

Comment:
Maybe we could change `infos` into a slightly less generic name such as `gitMetadata` or `gitInfo`?